spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2722,7 +2721,10 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
       case op
           if isCometSink(op) && op.output.forall(a =>
-            supportedDataType(a.dataType, allowComplex = true)) =>
+            supportedDataType(
+              a.dataType,
+              allowComplex =
+                usingDataFusionParquetExec(conf) || op.isInstanceOf[CometSparkToColumnarExec])) =>
```

Comment:
@mbutrovich I think this check require attention. 
It needs to work for `CometSparkToColumnarExec` as well